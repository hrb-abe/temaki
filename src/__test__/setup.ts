import * as Enzyme from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import * as ContextHook from 'babel-plugin-require-context-hook/register'

Enzyme.configure({ adapter: new Adapter() })
ContextHook()
