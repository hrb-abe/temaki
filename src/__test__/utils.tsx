import { mount, ReactWrapper, shallow, ShallowWrapper } from 'enzyme'
import * as React from 'react'
import { defaultTheme, ThemeProvider } from '~/modules/theme'

// 推論で型エラーがでるので、型定義してキャストする
type ShallowWithTheme = (tree: React.ReactElement) => ShallowWrapper
export const shallowWithTheme: ShallowWithTheme = tree => {
  return shallow(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>)
}

// 推論で型エラーがでるので、型定義してキャストする
type MountWithTheme = (tree: React.ReactElement) => ReactWrapper
export const mountWithTheme: MountWithTheme = tree => {
  return mount(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>)
}
