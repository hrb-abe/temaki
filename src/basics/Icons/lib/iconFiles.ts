import AddIcon from 'raw-loader!~/assets/icons/add.svg'
import AnalyzeIcon from 'raw-loader!~/assets/icons/analyze.svg'
import ArrowDown from 'raw-loader!~/assets/icons/arrow-down.svg'
import Attachment from 'raw-loader!~/assets/icons/attachment.svg'
import Avatar from 'raw-loader!~/assets/icons/avatar.svg'
import BlackBoard from 'raw-loader!~/assets/icons/blackbaord.svg'
import Calendar from 'raw-loader!~/assets/icons/calendar.svg'
import CheckBoxAll from 'raw-loader!~/assets/icons/checkBox-all.svg'
import CheckBoxOff from 'raw-loader!~/assets/icons/checkbox-off.svg'
import CheckBoxOn from 'raw-loader!~/assets/icons/checkbox-on.svg'
import ChevronDown from 'raw-loader!~/assets/icons/chevron-down.svg'
import ChevronLeft from 'raw-loader!~/assets/icons/chevron-left.svg'
import ChevronRight from 'raw-loader!~/assets/icons/chevron-right.svg'
import ChevronUp from 'raw-loader!~/assets/icons/chevron-up.svg'
import Close from 'raw-loader!~/assets/icons/close.svg'
import Company from 'raw-loader!~/assets/icons/company.svg'
import Copy from 'raw-loader!~/assets/icons/copy.svg'
import Delete from 'raw-loader!~/assets/icons/delete.svg'
import Download from 'raw-loader!~/assets/icons/download.svg'
import Drag from 'raw-loader!~/assets/icons/drag.svg'
import Dragdrop from 'raw-loader!~/assets/icons/dragdrop.svg'
import DropdownOff from 'raw-loader!~/assets/icons/dropdown-off.svg'
import DropdownOn from 'raw-loader!~/assets/icons/dropdown-on.svg'
import Edit from 'raw-loader!~/assets/icons/edit.svg'
import Event from 'raw-loader!~/assets/icons/event.svg'
import Excel from 'raw-loader!~/assets/icons/excel.svg'
import FilterApply from 'raw-loader!~/assets/icons/filter-apply.svg'
import Filters from 'raw-loader!~/assets/icons/filters.svg'
import GlobalMenu from 'raw-loader!~/assets/icons/global-menu.svg'
import Group from 'raw-loader!~/assets/icons/group.svg'
import Hamburger from 'raw-loader!~/assets/icons/hamburger.svg'
import Home from 'raw-loader!~/assets/icons/home.svg'
import Index from 'raw-loader!~/assets/icons/index.svg'
import Information from 'raw-loader!~/assets/icons/information.svg'
import Key from 'raw-loader!~/assets/icons/key.svg'
import Like from 'raw-loader!~/assets/icons/like.svg'
import Link from 'raw-loader!~/assets/icons/link.svg'
import List from 'raw-loader!~/assets/icons/list.svg'
import Lock from 'raw-loader!~/assets/icons/lock.svg'
import MenuV from 'raw-loader!~/assets/icons/menu-v.svg'
import MoveDown from 'raw-loader!~/assets/icons/move-down.svg'
import MoveUp from 'raw-loader!~/assets/icons/move-up.svg'
import Notifiacation from 'raw-loader!~/assets/icons/notification.svg'
import Phone from 'raw-loader!~/assets/icons/phone.svg'
import Pin from 'raw-loader!~/assets/icons/pin.svg'
import RadioOff from 'raw-loader!~/assets/icons/radio-off.svg'
import RadioOn from 'raw-loader!~/assets/icons/radio-on.svg'
import Redo from 'raw-loader!~/assets/icons/redo.svg'
import Refresh from 'raw-loader!~/assets/icons/refresh.svg'
import Remove from 'raw-loader!~/assets/icons/remove.svg'
import Search from 'raw-loader!~/assets/icons/search.svg'
import Settings from 'raw-loader!~/assets/icons/settings.svg'
import SortList from 'raw-loader!~/assets/icons/sort-list.svg'
import SortUpDown from 'raw-loader!~/assets/icons/sort-updown.svg'
import Sort from 'raw-loader!~/assets/icons/sort.svg'
import SortItems from 'raw-loader!~/assets/icons/sortItems.svg'
import Staff from 'raw-loader!~/assets/icons/staff.svg'
import Tag from 'raw-loader!~/assets/icons/tag.svg'
import Text from 'raw-loader!~/assets/icons/text.svg'
import Time from 'raw-loader!~/assets/icons/time.svg'
import Transition from 'raw-loader!~/assets/icons/transition.svg'
import TreeToggle from 'raw-loader!~/assets/icons/tree-toggle.svg'
import Undo from 'raw-loader!~/assets/icons/undo.svg'
import Url from 'raw-loader!~/assets/icons/url.svg'
import Warning from 'raw-loader!~/assets/icons/warning.svg'

export const icons = {
    AddIcon,
    AnalyzeIcon,
    ArrowDown,
    Attachment,
    Avatar,
    BlackBoard,
    Calendar,
    CheckBoxAll,
    CheckBoxOn,
    CheckBoxOff,
    ChevronUp,
    ChevronDown,
    ChevronLeft,
    ChevronRight,
    Close,
    Company,
    Copy,
    Delete,
    Download,
    Drag,
    Dragdrop,
    DropdownOff,
    DropdownOn,
    Edit,
    Event,
    Excel,
    FilterApply,
    Filters,
    GlobalMenu,
    Group,
    Hamburger,
    Home,
    Index,
    Information,
    Key,
    Like,
    Link,
    List,
    Lock,
    MenuV,
    MoveDown,
    MoveUp,
    Notifiacation,
    Phone,
    Pin,
    RadioOff,
    RadioOn,
    Redo,
    Refresh,
    Remove,
    Search,
    Settings,
    Sort,
    SortList,
    SortUpDown,
    SortItems,
    Staff,
    Tag,
    Text,
    Time,
    Transition,
    TreeToggle,
    Undo,
    Url,
    Warning
}
