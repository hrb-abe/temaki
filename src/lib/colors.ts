export default {
  primary: '#00A85F'
}
