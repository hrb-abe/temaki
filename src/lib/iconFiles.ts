import AddIcon from 'raw-loader!~/../icons/add.svg'
import AnalyzeIcon from 'raw-loader!~/../icons/analyze.svg'
import ArrowDown from 'raw-loader!~/../icons/arrow-down.svg'
import Attachment from 'raw-loader!~/../icons/attachment.svg'
import Avatar from 'raw-loader!~/../icons/avatar.svg'
import BlackBoard from 'raw-loader!~/../icons/blackbaord.svg'
import Calendar from 'raw-loader!~/../icons/calendar.svg'
import CheckBoxAll from 'raw-loader!~/../icons/checkBox-all.svg'
import CheckBoxOff from 'raw-loader!~/../icons/checkbox-off.svg'
import CheckBoxOn from 'raw-loader!~/../icons/checkbox-on.svg'
import ChevronDown from 'raw-loader!~/../icons/chevron-down.svg'
import ChevronLeft from 'raw-loader!~/../icons/chevron-left.svg'
import ChevronRight from 'raw-loader!~/../icons/chevron-right.svg'
import ChevronUp from 'raw-loader!~/../icons/chevron-up.svg'
import Close from 'raw-loader!~/../icons/close.svg'
import Company from 'raw-loader!~/../icons/company.svg'
import Copy from 'raw-loader!~/../icons/copy.svg'
import Delete from 'raw-loader!~/../icons/delete.svg'
import Download from 'raw-loader!~/../icons/download.svg'
import Drag from 'raw-loader!~/../icons/drag.svg'
import Dragdrop from 'raw-loader!~/../icons/dragdrop.svg'
import DropdownOff from 'raw-loader!~/../icons/dropdown-off.svg'
import DropdownOn from 'raw-loader!~/../icons/dropdown-on.svg'
import Edit from 'raw-loader!~/../icons/edit.svg'
import Event from 'raw-loader!~/../icons/event.svg'
import Excel from 'raw-loader!~/../icons/excel.svg'
import FilterApply from 'raw-loader!~/../icons/filter-apply.svg'
import Filters from 'raw-loader!~/../icons/filters.svg'
import GlobalMenu from 'raw-loader!~/../icons/global-menu.svg'
import Group from 'raw-loader!~/../icons/group.svg'
import Hamburger from 'raw-loader!~/../icons/hamburger.svg'
import Home from 'raw-loader!~/../icons/home.svg'
import Index from 'raw-loader!~/../icons/index.svg'
import Information from 'raw-loader!~/../icons/information.svg'
import Key from 'raw-loader!~/../icons/key.svg'
import Like from 'raw-loader!~/../icons/like.svg'
import Link from 'raw-loader!~/../icons/link.svg'
import List from 'raw-loader!~/../icons/list.svg'
import Lock from 'raw-loader!~/../icons/lock.svg'
import MenuV from 'raw-loader!~/../icons/menu-v.svg'
import MoveDown from 'raw-loader!~/../icons/move-down.svg'
import MoveUp from 'raw-loader!~/../icons/move-up.svg'
import Notifiacation from 'raw-loader!~/../icons/notification.svg'
import Phone from 'raw-loader!~/../icons/phone.svg'
import Pin from 'raw-loader!~/../icons/pin.svg'
import RadioOff from 'raw-loader!~/../icons/radio-off.svg'
import RadioOn from 'raw-loader!~/../icons/radio-on.svg'
import Redo from 'raw-loader!~/../icons/redo.svg'
import Refresh from 'raw-loader!~/../icons/refresh.svg'
import Remove from 'raw-loader!~/../icons/remove.svg'
import Search from 'raw-loader!~/../icons/search.svg'
import Settings from 'raw-loader!~/../icons/settings.svg'
import SortList from 'raw-loader!~/../icons/sort-list.svg'
import SortUpDown from 'raw-loader!~/../icons/sort-updown.svg'
import Sort from 'raw-loader!~/../icons/sort.svg'
import SortItems from 'raw-loader!~/../icons/sortItems.svg'
import Staff from 'raw-loader!~/../icons/staff.svg'
import Tag from 'raw-loader!~/../icons/tag.svg'
import Text from 'raw-loader!~/../icons/text.svg'
import Time from 'raw-loader!~/../icons/time.svg'
import Transition from 'raw-loader!~/../icons/transition.svg'
import TreeToggle from 'raw-loader!~/../icons/tree-toggle.svg'
import Undo from 'raw-loader!~/../icons/undo.svg'
import Url from 'raw-loader!~/../icons/url.svg'
import Warning from 'raw-loader!~/../icons/warning.svg'

export const icons = {
    AddIcon,
    AnalyzeIcon,
    ArrowDown,
    Attachment,
    Avatar,
    BlackBoard,
    Calendar,
    CheckBoxAll,
    CheckBoxOn,
    CheckBoxOff,
    ChevronUp,
    ChevronDown,
    ChevronLeft,
    ChevronRight,
    Close,
    Company,
    Copy,
    Delete,
    Download,
    Drag,
    Dragdrop,
    DropdownOff,
    DropdownOn,
    Edit,
    Event,
    Excel,
    FilterApply,
    Filters,
    GlobalMenu,
    Group,
    Hamburger,
    Home,
    Index,
    Information,
    Key,
    Like,
    Link,
    List,
    Lock,
    MenuV,
    MoveDown,
    MoveUp,
    Notifiacation,
    Phone,
    Pin,
    RadioOff,
    RadioOn,
    Redo,
    Refresh,
    Remove,
    Search,
    Settings,
    Sort,
    SortList,
    SortUpDown,
    SortItems,
    Staff,
    Tag,
    Text,
    Time,
    Transition,
    TreeToggle,
    Undo,
    Url,
    Warning
}
