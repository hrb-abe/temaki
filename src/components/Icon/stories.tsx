import { color, text } from '@storybook/addon-knobs'
import * as Storybook from '@storybook/react'
import * as React from 'react'

import * as IconFiles from '~/lib/iconFiles'
import Icon from './index'

Storybook.storiesOf('components|Icon', module).add('Standard', () => (
    <Icon
        svg={IconFiles.icons.AddIcon}
        size={text('Size', '24px')}
        color={color('Color', '#333')}
    />
))
