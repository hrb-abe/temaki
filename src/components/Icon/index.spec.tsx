import * as Enzyme from 'enzyme'
import * as React from 'react'

import * as IconFiles from '~/basics/Icons/lib/iconFiles'
import { mountWithTheme } from '../../__test__/utils'
import Icon from './index'

describe('Icon', () => {
  let wrapper: Enzyme.ReactWrapper

  beforeEach(() => {
    wrapper = mountWithTheme(<Icon svg={IconFiles.icons.CheckBoxAll} size="24px" color="#aaa" />)
  })

  it('should be defined', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('should match before snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
