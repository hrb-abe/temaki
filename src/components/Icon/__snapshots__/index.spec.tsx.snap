// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match before snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "grayScale": Object {
          "S0": "rgb(255, 255, 255)",
          "S10": "rgb(234, 234, 234)",
          "S100": "rgb(51, 51, 51)",
          "S20": "rgb(214, 214, 214)",
          "S30": "rgb(193, 193, 193)",
          "S40": "rgb(173, 173, 173)",
          "S5": "rgb(245, 245, 245)",
          "S50": "rgb(153, 153, 153)",
          "S60": "rgb(133, 133, 133)",
          "S70": "rgb(112, 112, 112)",
          "S80": "rgb(92, 92, 92)",
          "S90": "rgb(71, 71, 71)",
        },
        "primary": Object {
          "N20": "rgb(92, 92, 92)",
          "N40": "rgb(133, 133, 133)",
          "N60": "rgb(173, 173, 173)",
          "N80": "rgb(214, 214, 214)",
          "N95": "rgb(245, 245, 245)",
          "P20": "rgb(41, 41, 41)",
          "P40": "rgb(31, 31, 31)",
          "P60": "rgb(20, 20, 20)",
          "P80": "rgb(10, 10, 10)",
          "P95": "rgb(0, 0, 0)",
          "default": "rgb(51, 51, 51)",
        },
        "text": Object {
          "default": "#333",
        },
        "utilities": Object {
          "blue": "rgb(48, 120, 191)",
          "highlightGreen": "rgb(114, 206, 92)",
          "paleBlue": "rgb(245, 249, 252)",
          "paleGreen": "rgb(248 ,252, 247)",
          "paleRed": "rgb(253, 246, 246)",
          "red": "rgb(224, 85, 72)",
        },
      },
      "shadows": Object {
        "L1": "0 0 2px 0 rgba(0, 0, 0, 0.16)",
        "L2": "0 1px 3px 0 rgba(0, 0, 0, 0.16)",
        "L3": "0 2px 5px 0 rgba(0, 0, 0, 0.16)",
        "L4": "0 3px 8px 0 rgba(0, 0, 0, 0.16)",
        "L5": "0 5px 11px 0 rgba(0, 0, 0, 0.16)",
      },
      "typography": Object {
        "pc": Object {
          "body": Object {
            "L": "16px",
            "S": "12px",
            "XL": "18px",
            "default": "14px",
          },
          "heading": Object {
            "H1": "24px",
            "H2": "22px",
            "H3": "20px",
            "H4": "18px",
            "H5": "16px",
            "H6": "14px",
          },
        },
        "sp": Object {
          "body": Object {
            "L": "16px",
            "S": "12px",
            "XL": "18px",
            "default": "14px",
          },
          "heading": Object {
            "H1": "24px",
            "H2": "22px",
            "H3": "20px",
            "H4": "18px",
            "H5": "16px",
            "H6": "14px",
          },
        },
      },
    }
  }
>
  <Component
    color="#aaa"
    size="24px"
  >
    <styled.div
      color="#aaa"
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
      size="24px"
    >
      <StyledComponent
        color="#aaa"
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "jxVDXv",
              "rules": Array [
                "
    height: ",
                [Function],
                ";
    width: ",
                [Function],
                ";
    & svg {
        height: ",
                [Function],
                ";
        width: ",
                [Function],
                ";
    }
    & #fill {
        fill: ",
                [Function],
                ";
    }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        size="24px"
      >
        <div
          className="sc-bdVaJa jxVDXv"
          color="#aaa"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          size="24px"
        />
      </StyledComponent>
    </styled.div>
  </Component>
</ThemeProvider>
`;
